aana/api/request_handler.py

```diff
@@ -229,3 +230,46 @@ async def _async_chat_completions(
                 "created": int(time.time()),
                 "choices": [{"index": 0, "message": response["message"]}],
             }
+
+    @app.get("/api/status", response_model=SDKStatusResponse)
+    async def status(self) -> SDKStatusResponse:
+        """The endpoint for checking the status of the application.
+
+        Returns:
+            SDKStatusResponse: The response containing the status of the application.
+        """
+        serve_status = serve.status()
+        message = ""
+        if any(
+            app.status == "DEPLOY_FAILED"
+            or app.status == "UNHEALTHY"
+            or app.status == "NOT_STARTED"
+            for app in serve_status.applications.values()
+        ):
+            sdk_status = SDKStatus.UNHEALTHY
+            error_messages = []
+            for app_name, app_status in serve_status.applications.items():
+                if (
+                    app_status.status == "DEPLOY_FAILED"
```

Comment:
Should we add the `NOT_STARTED` apps included in message too?

Reply:
What do you mean by "default message of app_status"?

Reply:
The ray serve status  returns something like this:
```json
applications:
  default:
    status: RUNNING
    message: ''"   <--- this one
    last_deployed_time_s: 1694041157.2211847
    deployments:
      Translator:
        status: HEALTHY
        replica_states:
          RUNNING: 1
        message: ''
      Summarizer:
        status: HEALTHY
        replica_states:
          RUNNING: 1
        message: ''
 ```

Reply:
It's empty if there isn't any error.